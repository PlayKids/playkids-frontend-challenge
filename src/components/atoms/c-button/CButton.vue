<template>
  <button @click="$emit('click')" v-bind="$attrs">
    <slot />
  </button>
</template>

<script>
export default {
  name: 'CButton'
};
</script>
