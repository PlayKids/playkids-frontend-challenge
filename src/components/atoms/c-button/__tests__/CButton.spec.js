import { mount } from '@vue/test-utils';
import CButton from '../CButton.vue';

describe('CButton component', () => {
  it('renders correctly', () => {
    const wrapper = mount(CButton);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls click method', async () => {
    const wrapper = mount(CButton);
    await wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
  });
});
