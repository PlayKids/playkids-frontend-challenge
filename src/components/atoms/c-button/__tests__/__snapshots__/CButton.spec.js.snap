// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButton component renders correctly 1`] = `<button class="button"></button>`;
